src/Microsoft.Azure.WebJobs.Host/Bindings/BindingProviders/FilteringBindingProvider.cs

```diff
@@ -26,7 +33,16 @@ public Task<IBinding> TryCreateAsync(BindingProviderContext context)
                 throw new ArgumentNullException("context");
             }
             Type parametertype = context.Parameter.ParameterType;
-            if (!_predicate(parametertype))
+
+            var attr = context.Parameter.GetCustomAttribute<TAttribute>();
+
+            var cloner = new AttributeCloner<TAttribute>(attr, _nameResolver);
+            var attrNameResolved = cloner.GetNameResolvedAttribute();
+
+            // This may do validation and throw too. 
+            bool canBind = _predicate(attrNameResolved, parametertype);
+
+            if (!canBind)
```

Comment:
If I add several filters in a row -- shouldn't it try all the filters and only return null if none of them returned true?

i.e. -- if i want to support TypeA, TypeB, TypeC, I would think that I could do:

AddFilter(IsTypeA)
AddFilter(IsTypeB)
AddFilter(IsTypeC)

If the type is TypeB, I'd expect it to match